sync/barrier_test.go

```diff
@@ -161,6 +161,29 @@ func TestBarrierDeadline(t *testing.T) {
 	}
 }
 
+func TestInmemSignalAndWait(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	runenv, cleanup := runtime.RandomTestRunEnv(t)
+	t.Cleanup(cleanup)
```

Comment:
Nice one, I wasn't aware of that. We could refactor in a future PR the `RandomTestRunEnv` to not return a `cleanup` function at all, and jus use `t.Cleanup` inside.